ymca/src/main/java/endpoints/LoggingEndpoint.java

```diff
@@ -0,0 +1,104 @@
+package endpoints;
+
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+import java.util.logging.Level;
+
+import javax.websocket.CloseReason;
+import javax.websocket.Endpoint;
+import javax.websocket.EndpointConfig;
+import javax.websocket.MessageHandler;
+import javax.websocket.Session;
+
+import org.json.JSONObject;
+
+/**
+ * Endpoint Class used to log messages and send them to the client
+ * @author Jake Lewis
+ *
+ */
+public class LoggingEndpoint extends Endpoint
+{
+
+	private static Map<String, Set<LoggingEndpoint>> endpoints = new HashMap<>();
+	private Session session;
+	private String sessionId;
+
+	@Override
+	public void onOpen(Session session, EndpointConfig config)
+	{
+		this.session = session;
+		session.addMessageHandler(new MessageHandler.Whole<String>()
+		{
+			@Override
+			public void onMessage(String text)
+			{
+				sessionId = text;
+			}
+		});
+
+		if (null == LoggingEndpoint.endpoints.get(sessionId))
+		{
+			LoggingEndpoint.endpoints.put(sessionId, new HashSet<LoggingEndpoint>());
+			LoggingEndpoint.endpoints.get(sessionId).add(this);
+		}
+	}
+
+	/**
+	 * Event that is triggered when a session has closed.
+	 *
+	 * @param session
+	 *            The session
+	 * @param closeReason
+	 *            Why the session was closed
+	 */
+	@Override
+	public void onClose(final Session session, final CloseReason closeReason)
+	{
+		LoggingEndpoint.endpoints.remove(session.getId());
```

Comment:
this should use the http session id (i.e. this.sessionId) not `session.getId()`) as websocket sessions are different to http sessions